src/librustc_trans/trans/cabi_x86_64.rs

```diff
@@ -410,18 +410,46 @@ pub fn compute_abi_info(ccx: &CrateContext,
         }
     }
 
-    let mut arg_tys = Vec::new();
-    for t in atys {
-        let ty = x86_64_ty(ccx, *t, |cls| cls.is_pass_byval(), Attribute::ByVal);
-        arg_tys.push(ty);
-    }
+    let mut avail_regs = 6; // RDI, RSI, RDX, RCX, R8, R9
```

Comment:
Yes but, this is a kinda simplistic approach similar to what clang does since we need to explicitly tell LLVM whether a struct is passed via some hidden pointer (marked byval) or split up/passed as a FCA. The actual specifics of which register or whether it goes on the stack is handled by LLVM.